### PR TITLE
XFA - Fix lot of layout issues

### PR DESCRIPTION
**Pull request by calixteman**
- I thought it was possible to rely on browser layout engine to handle layout stuff but it isn't possible
    - mainly because when a contentArea overflows, we must continue to layout in the next contentArea
    - when no more contentArea is available then we must go to the next page...
    - we must handle breakBefore and breakAfter which allows to "break" the layout to go to the next container
  - Sometimes some containers don't provide their dimensions so we must compute them in order to know where to put
    them in their parents but to compute those dimensions we need to layout the container itself...
  - See top of file layout.js for more explanations about layout.
  - fix few bugs in other places I met during my work on layout.

### CODE REVIEW

File: src/core/xfa/layout.js [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { $extra, $flushHTML } from "./xfa_object.js";
+import { measureToString } from "./html_utils.js";
+
+// Subform and ExclGroup have a layout so they share these functions.
+
+/*
```
Comment (Snuffleupagus) [inline src/core/xfa/layout.js:21]: Nit: Please remember to use the following comment formatting, for overall consistency in the code-base.
```javascript 
/**
 * How layout works?
 *
 * ...
 */
```

File: src/core/xfa/template.js [hunk 2]
```diff
@@ -1683,29 +1925,154 @@ class ExclGroup extends XFAObject {
...
+    attributes.style = style;
+    attributes.class = clazz.join(" ");
+
+    if (this.name) {
+      attributes.xfaName = this.name;
+    }
+
+    let failure;
+    if (this.layout === "lr-tb" || this.layout === "rl-tb") {
+      for (; this[$extra].attempt < 3; this[$extra].attempt++) {
```
Comment (brendandahl) [inline src/core/xfa/template.js:2020]: Make number of attempts a constant.

**Review by brendandahl — CHANGES_REQUESTED** [review]

**Review by brendandahl — APPROVED** [review]
r+ needs the comment in layout.js updated though.

**Pull request merged**